### PR TITLE
Implemented an option to change language

### PR DESCRIPTION
**Pull request by mteiting**
## Reason for PullRequest
Implemented a feature which allows you to change the language, as there is no such feature yet.

## Description of the Changes
Added a config entry in the App section of the configfile and a new section in the properties, to be able to change the language during runtime.

## Notes
In order to use the selected language the app has to be restarted.

### CODE REVIEW

**Review by X39 — CHANGES_REQUESTED** [review]
Missing a way to notify the user about required restart

**Comment by X39** [comment]
to get a callback, a wrapper item can be created
example:
```cs
using System;

namespace ArmA.Studio.Data.Configuration
{
    public class ChangeCallbackItemWrapper : Item
    {
        public Item InnerItem { get; private set; }
        public Func<object, object> Callback { get; private set; }

        public ChangeCallbackItemWrapper(Item i, Action<object> callback) : this(i, (obj) => { callback(obj); return obj; }) { }
        public ChangeCallbackItemWrapper(Item i, Func<object, object> callback) : base(i.Template, i.Value)
        {
            this.InnerItem = i;
            this.Callback = callback;
        }


        public override bool IsValidValue(object value) => this.InnerItem.IsValidValue(value);

        public override object OnValueChange(object oldValue, object newValue)
        {
            var val = this.InnerItem.OnValueChange(oldValue, newValue);
            val = this.Callback.Invoke(val);
            return val;
        }
    }
}
```

**Comment by mteiting** [comment]
Implemented dialog which notifies the user and asks him if he wants to restart the app right away

File: ArmA.Studio.Data/Configuration/ComboBoxItem.cs [hunk 1]
```diff
@@ -13,15 +13,34 @@ public class ComboBoxItem<T> : PropertyItem
         private readonly static DataTemplate ThisDataTemplate = LoadFromEmbeddedResource<DataTemplate>(typeof(ComboBoxItem<T>).Assembly, "ArmA.Studio.Data.Configuration.ComboBoxItem.xaml");
         public IEnumerable<KeyValuePair<string, T>> KeyValueCollection { get; private set; }
 
-        public ComboBoxItem(IEnumerable<KeyValuePair<string, T>> values, string name, string path, object propertyOwner) : this(values, name, string.Empty, propertyOwner.GetType().GetProperty(path), propertyOwner) { }
-        public ComboBoxItem(IEnumerable<KeyValuePair<string, T>> values, string name, string icon, string path, object propertyOwner) : this(values, name, icon, propertyOwner.GetType().GetProperty(path), propertyOwner) { }
-        public ComboBoxItem(IEnumerable<KeyValuePair<string, T>> values, string name, PropertyInfo property, object propertyOwner) : this(values, name, string.Empty, property, propertyOwner) { }
+        public Func<object, object> ValueChangeCallback { get; set; } = null;
```
Comment (X39) [inline ArmA.Studio.Data/Configuration/ComboBoxItem.cs]: it is not supposed to add value changed callbacks to Items

use property getter & setter or proxy properties for this

`public object proxyproperty { get { return origin.property; } set { /* handle required stuff for on-change */ origin.property = value } }`

File: ArmA.Studio/Dialogs/PropertiesDialogDataContext.cs [hunk 2]
```diff
@@ -81,11 +83,33 @@ private IEnumerable<SubCategory> GetGeneralSubCategories()
...
             { Name = Properties.Localization.Properties_General_Language, ImageSource = @"/ArmA.Studio;component/Resources/Logo.ico" };
         }
 
+        private object LanguageChanged( object o )
+        {
+            var msgbox = MessageBox.Show( Properties.Localization.MessageDialog_PropertyChange, "Property", MessageBoxButton.YesNo );
+
+            if ( msgbox == MessageBoxResult.Yes )
+            {
+                Application.Current.Exit += ( sender, args ) =>
```
Comment (X39) [inline ArmA.Studio/Dialogs/PropertiesDialogDataContext.cs]: Please use static function `App.Shutdown(...)` for this

**Review by X39 — CHANGES_REQUESTED** [review]

**Comment by mteiting** [comment]
Implemented a seperate LanguageManager which handles the logic.

**Review by X39 — APPROVED** [review]

**Pull request merged**